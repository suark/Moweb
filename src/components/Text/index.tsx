import { useTheme } from '@react-navigation/native';
import React from 'react';
import { Text as DefaultText, TextProps } from 'react-native';

export const Text = (props: TextProps) => {
  const { style, ...otherProps } = props;
  const { colors } = useTheme();

  return (
    <DefaultText
      style={[{ color: colors.text }, style]}
      {...otherProps}
    />
  );
};
