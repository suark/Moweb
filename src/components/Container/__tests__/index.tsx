import React from 'react';
import renderer from 'react-test-renderer';
import { Container } from '..';
import { Text } from '../../Text';

it('should render correctly', () => {
  const component = renderer.create(
    <Container testID='wrapper'>
      <Text>hello</Text>
    </Container>,
  );

  expect(component.toJSON()).toMatchSnapshot();
});
