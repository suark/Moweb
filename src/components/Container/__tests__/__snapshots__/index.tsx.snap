// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="wrapper"
>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(28, 28, 30)",
        },
        undefined,
      ]
    }
  >
    hello
  </Text>
</View>
`;
